util/jsonutil/keywordsutil.go

```diff
@@ -0,0 +1,59 @@
+package jsonutil
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+type KeywordsString string
+
+// extImpAppnexusKeyVal defines the contract for bidrequest.imp[i].ext.prebid.bidder.appnexus.keywords[i]
+type extImpAppnexusKeyVal struct {
+	Key    string   `json:"key,omitempty"`
+	Values []string `json:"value,omitempty"`
+}
+
+func (ks *KeywordsString) UnmarshalJSON(b []byte) error {
+	if len(b) == 0 {
+		return nil
+	}
+
+	switch b[0] {
+	case '{':
+		var results map[string][]string
+		err := json.Unmarshal(b, &results)
+		if err != nil {
+			return err
+		}
+
+		var keywords string
+		for key, values := range results {
+			for _, val := range values {
+				keywords = keywords + fmt.Sprintf("%s=%s,", key, val)
```

Comment:
Please use a `strings.Builder` for efficiency instead of concatenating to a string in a loop. Same for `kvs` in the other case.

Reply:
addressed in [d97583d](https://github.com/prebid/prebid-server/pull/2649/commits/d97583d4629a56be5f7fbe6a6e78c741caedf366)